sintr_analysis_server_example/lib/query/perf.dart

```diff
@@ -1,71 +1,220 @@
-library sintr_worker_lib.instrumentation_lib;
+// Copyright (c) 2015, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library sintr_worker_lib.query.perf;
 
 import 'dart:convert';
 
-export '../instrumentation_processor.dart';
+import 'package:sintr_worker_lib/instrumentation_query.dart';
 
-_sanityCheck(String ln) {
-  //~1422655684588:Noti:
-  //01234567890123456789
+const ANALYSIS_PERF = 'Analysis';
+const INTERNAL_PERF = 'Internal';
+const REQUEST_PERF = 'Request';
 
-  List<String> splits = ln.split(":");
-  String msgType = splits[1];
+/// Add an extraction result to the overall API usage [results]
```

Comment:
I'm trying to estimate overall usage of each Analysis Server request type. In some cases such as setAnalysisRoots, it does not matter, but it would be good to know if some APIs are not used by any clients. This is a first cut as additional information should be collected. For example this captures the number of times refactoring is called, but it would be great to capture which refactorings are being chosen. As always, suggestions are most welcome.